### PR TITLE
#9: Configure a single Concordium account in the backend (via env file) to mint NFTs with it

### LINKED ISSUE #9: Configure a single Concordium account in the backend (via env file) to mint NFTs with it

### PR DESCRIPTION
**Pull request by CongThuong**

### CODE REVIEW

File: api/controllers/productController.js [hunk 1]
```diff
@@ -13,6 +13,9 @@ const multer = require("multer");
 let dirImg   = '/image/';
 
 exports.add = async (req, res) => {
+  const token = await concordium.getNextNFT();
+
+  const trx   = await concordium.mintNFT("4kF3psLpaLYLZXLpPinLUFe2ytMhLsyzNoVCKACE8a6BoPhiZ3", token);
```
Comment (jatitoam) [inline api/controllers/productController.js]: Please make sure the wallet is in the env file

File: api/web3/concordium.js [hunk 2]
```diff
@@ -49,12 +51,45 @@ class Concordium {
...
+  async getNextNFT() {
+    return String(Object.keys(await this.listNFTs()).length + 1).padStart(
+        8,
+        "0"
+    );
+  }
+  async mintNFT(address, token) {
+    return await this.client.sendTransaction(
+        address,
+        AccountTransactionType.Update,
```
Comment (jatitoam) [inline api/web3/concordium.js]: Is there no need for the wallet private key to authorize the transaction?  Can I use any wallet and have it spend money? There is something missing here

File: api/controllers/productController.js [hunk 3]
```diff
@@ -13,6 +13,8 @@ const multer = require("multer");
 let dirImg   = '/image/';
 
 exports.add = async (req, res) => {
+  const token = await concordium.getNextNFT();
```
Comment (jatitoam) [inline api/controllers/productController.js:16]: now that you have the token here, just make sure the token and block are used in https://github.com/aesirxio/web3-demo-backend/pull/8/files#diff-ccf008b2097c4d1961dd9af98911474b941d37680598698ee1ab84bd0e1ad993R34

File: api/web3/concordium.js [hunk 4]
```diff
@@ -49,12 +55,76 @@ class Concordium {
...
+    const nonce            = nextAccountNonce.nonce;
+    const signer           = await buildBasicAccountSigner(this.private_key);
+
+    const contractName        = this.contractName;
+    const receiveFunctionName = "mint";
+    const receiveName         = contractName + '.' + receiveFunctionName;
+    const rawModuleSchema     = Buffer.from(fs.readFileSync('./schema.bin'));
+    const schemaVersion       = SchemaVersion.V1;
+    const parameters          = {
+      owner: {Account: [this.address]},
```
Comment (jatitoam) [inline api/web3/concordium.js]: super important: this account here needs to be received (and validated with signature) via API, so the API will define who the owner of this token will be

Comment (CongThuong) [reply api/web3/concordium.js]: Please take a look again @jatitoam

**Review by jatitoam — APPROVED** [review]

**Pull request merged**